### PR TITLE
Added Datepicker for +10 supply button

### PR DESCRIPTION
**Pull request by diliger**
d3 timeline fixed to resize according to parent container

### CODE REVIEW

**Comment by diliger** [comment]
Hi @tomalec !

Can you please review my pull request when possible..

File: src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html [hunk 1]
```diff
@@ -38,16 +42,24 @@
...
                 this.attachedFired = true;
-                this.width = this.width || 600;
-                this.height = this.height || 250;
-                this.footerHeight = this.footerHeight || 150;
-                this.innerWidth = this.width - 40;
-                this.innerHeight = this.height - 40;
-                this.graphHeight = this.innerHeight - this.footerHeight;
 
                 setTimeout(function () {
+                    var size = this.getBoundingClientRect();
```
Comment (tomalec) [inline src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html]: Could you make it more DRY with what's in `updateSize`?

File: src/Products/wwwroot/Products/viewmodels/EditProductPage.html [hunk 2]
```diff
@@ -28,7 +29,11 @@
                         <tr>
                             <td>Shop</td>
                             <td class="products-shops-box__triangle">&#9660;</td>
-                            <td></td>
+                            <td>
+                                <pikaday-decorator format="{{model.ClientDateFormat}}">
+                                    <input type="text" class="form-control" value="{{model.SupplyDate$::change}}" placeholder="Supply date" style="width:40px; height: 25px;"/>
```
Comment (tomalec) [inline src/Products/wwwroot/Products/viewmodels/EditProductPage.html]: Could you move inline styles to the `products.css`?

File: src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
-            <span>{{model.DateTo}}</span>
+            <pikaday-decorator format="{{model.ClientDateFormat}}">
+                <input type="text" class="" value="{{model.DateTo$::change}}" placeholder="Date to" />
+            </pikaday-decorator>
         </h5>
         <button class="btn btn-primary" onmousedown="++this.value;" value="{{model.Event$::click}}" slot="Products/event">
             Test event
         </button>
-        <timeline-element
+        <timeline-element style="height:350px;"
```
Comment (tomalec) [inline src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html]: ..and this as well.

File: src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html (hunk 1 above)
Comment (diliger) [inline src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html]: `updateSize` is a function to re-render the graph according to container dimensions,
it's used on window.resize

**Comment by diliger** [comment]
What dependency to add to bower.json?
Are you talking about timeline-element? 
It is implemented directly inside Products app, so no separate repo in github..

File: src/Products/wwwroot/Products/viewmodels/EditProductPage.html (hunk 2 above)
Comment (diliger) [inline src/Products/wwwroot/Products/viewmodels/EditProductPage.html]: fixed.

File: src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html (hunk 3 above)
Comment (diliger) [inline src/Products/wwwroot/Products/viewmodels/ProductSalesPage.html]: fixed.

**Comment by warpech** [comment]
@joozek78 could you please finish the review? Probably it is fine after recent fixes

File: src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html (hunk 1 above)
Comment (joozek78) [inline src/Products/wwwroot/Products/elements/d3-timeline/timeline-element.html]: Well, there's obviously a lot of similarities. I'd also go for drying it up. Or are the similarities incidental?

File: src/Products/wwwroot/Products/viewmodels/EditProductPage.html (hunk 2 above)
Comment (joozek78) [inline src/Products/wwwroot/Products/viewmodels/EditProductPage.html]: It doesn't look like :(

**Pull request merged**